File: backend/core/events/domain/entity/event_place.ts
/**
 * イベント会場エンティティ
 * event_places テーブルを表現
 */
export interface EventPlacePayload {
  id: string
  name: string
  googleMapsUrl: string
  createdAt: Date
  createdUser: string | null
  updatedAt: Date
  updatedUser: string | null
}

export class EventPlace {
  private constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly googleMapsUrl: string,
    public readonly createdAt: Date,
    public readonly createdUser: string | null,
    public readonly updatedAt: Date,
    public readonly updatedUser: string | null,
  ) {}

  static fromPayload(payload: EventPlacePayload): EventPlace {
    return new EventPlace(
      payload.id,
      payload.name,
      payload.googleMapsUrl,
      payload.createdAt,
      payload.createdUser,
      payload.updatedAt,
      payload.updatedUser,
    )
  }

  toResponse(): {
    id: string
    name: string
    googleMapsUrl: string
  } {
    return {
      id: this.id,
      name: this.name,
      googleMapsUrl: this.googleMapsUrl,
    }
  }
}

File: apps/mobile/lib/event_create/models/event_place.dart
class EventPlace {
  const EventPlace({
    required this.id,
    required this.name,
    required this.googleMapsUrl,
    required this.createdAt,
    this.createdUser,
    required this.updatedAt,
    this.updatedUser,
  });

  factory EventPlace.fromJson(Map<String, dynamic> json) {
    return EventPlace(
      id: json['id'] as String,
      name: json['name'] as String,
      googleMapsUrl: json['google_maps_url_normalized'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
      createdUser: json['created_user'] as String?,
      updatedAt: DateTime.parse(json['updated_at'] as String),
      updatedUser: json['updated_user'] as String?,
    );
  }

  final String id;
  final String name;
  final String googleMapsUrl; // DBカラム名: google_maps_url_normalized
  final DateTime createdAt;
  final String? createdUser;
  final DateTime updatedAt;
  final String? updatedUser;
}
